File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem01_MergeStringsAlternately.java
package org.phoenix.leetcode.challenges;

public class Problem01_MergeStringsAlternately {

    public String mergeAlternately(String word1, String word2) {
        StringBuilder str = new StringBuilder();
        int m = word1.length();
        int n = word2.length();
        int i = 0;
        int j = 0;
        
        while (i < m || j < n) {
            if (i < m) str.append(word1.charAt(i++));
            if (j < n) str.append(word2.charAt(j++));
        }
        return str.toString();
    }
}

File: JavaScript/leetcode75/src/main/Problem01_MergeStringsAlternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function(word1, word2) {
    let i =0;
    let j =0;
    let arr = [];
    let m = word1.length;
    let n = word2.length;
    while(i < m || j < n) {
        if(i<m) arr.push(word1[i++]);
        if(j<n) arr.push(word2[j++]);
    }
    return arr.join('');
};

module.exports = mergeAlternately;